recipes/swig/all/conanfile.py

```diff
@@ -1,170 +1,164 @@
             autotools.make()
 
     def package(self):
-        self.copy(pattern="LICENSE*", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="COPYRIGHT", dst="licenses", src=self._source_subfolder)
-        self.copy("*", src="cmake", dst=self._module_subfolder)
-        with self._build_context():
-            autotools = self._configure_autotools()
+        copy(self, "LICENSE*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, "COPYRIGHT", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, "*.cmake",
+             dst=os.path.join(self.package_folder, self._module_subfolder),
+             src=os.path.join(self.export_sources_folder, "cmake"))
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
             autotools.install()
-
-    @property
-    def _swiglibdir(self):
-        return os.path.join(self.package_folder, "bin", "swiglib").replace("\\", "/")
+        for path in self.package_path.iterdir():
+            if path.is_dir() and path.name not in ["bin", "lib", "licenses"]:
+                rmdir(self, path)
 
     @property
     def _module_subfolder(self):
         return os.path.join("lib", "cmake")
 
     @property
-    def _module_file(self):
-        return "conan-official-{}-targets.cmake".format(self.name)
+    def _cmake_module_rel_path(self):
+        return os.path.join(self._module_subfolder, "conan-swig-variables.cmake")
 
     def package_info(self):
-        self.cpp_info.includedirs=[]
+        self.cpp_info.includedirs = []
+        self.cpp_info.set_property("cmake_file_name", "SWIG")
+        self.cpp_info.set_property("cmake_target_name", "SWIG::SWIG")
+        self.cpp_info.set_property("cmake_build_modules", [self._cmake_module_rel_path])
+
+        self.buildenv_info.define_path("SWIG_LIB", os.path.join(self.package_folder, "bin", "swiglib"))
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
         self.cpp_info.names["cmake_find_package"] = "SWIG"
         self.cpp_info.names["cmake_find_package_multi"] = "SWIG"
-        self.cpp_info.builddirs = [self._module_subfolder]
-        self.cpp_info.build_modules["cmake_find_package"] = \
-            [os.path.join(self._module_subfolder, self._module_file)]
-        self.cpp_info.build_modules["cmake_find_package_multi"] = \
-            [os.path.join(self._module_subfolder, self._module_file)]
+        self.cpp_info.build_modules["cmake_find_package"] = [self._cmake_module_rel_path]
+        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._cmake_module_rel_path]
 
         bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bindir))
         self.env_info.PATH.append(bindir)
+        self.env_info.SWIG_LIB = os.path.join(self.package_folder, "bin", "swiglib")
```

Comment:
For conan v2 do we want `runenv_info` too?

```suggestion
        self.runenv_info.prepend_path("PATH", bindir)
        self.env_info.SWIG_LIB = os.path.join(self.package_folder, "bin", "swiglib")
```